org.eclipse.ice.dev/org.eclipse.ice.dev.dependencyscraper/src/main/java/org/org/eclipse/ice/dev/dependencyscraper/DependencyScraper.java

```diff
@@ -0,0 +1,227 @@
+
+	/**
+	 * Whether files should be overwritten if they already exist.
+	 */
+	@Parameter(
+		property = "clobber",
+		required = false,
+		defaultValue = "false"
+	)
+	private boolean clobber;
+
+	/**
+	 * Set of jar files that will be searched for matching files.
+	 */
+	private Set<File> jarFiles;
+
+	/**
+	 * Setter for includes.
+	 * @param includes to set.
+	 */
+	public void setIncludes(List<String> includes) {
+		this.includes = includes;
+	}
+
+	/**
+	 * Setter for clobber.
+	 * @param clobber to set.
+	 */
+	public void setClobber(boolean clobber) {
+		this.clobber = clobber;
+	}
+
+	/**
+	 * Setter for jarFiles.
+	 * @param jarFiles to set.
+	 */
+	public void setJarFiles(Set<File> jarFiles) {
+		this.jarFiles = jarFiles;
+	}
+
+	/**
+	 * Filter for whether a ZipEntry begins with the sourceDirectory.
+	 * @param file to test
+	 * @return if ZipEntry path begins with sourceDirectory.
+	 */
+	private boolean startsWithSourceDirectory(ZipEntry file) {
+		return file.getName()
+			.toLowerCase(Locale.ENGLISH)
+			.startsWith(sourceDirectory.toLowerCase(Locale.ENGLISH));
+	}
+
+	/**
+	 * Filter for whether a ZipEntry should be included.
+	 * @param file to test
+	 * @return if it should be included
+	 */
+	private boolean shouldInclude(ZipEntry file) {
+		return includes.stream()
+			.anyMatch(
+				include -> FilenameUtils.wildcardMatch(file.getName(), include)
+			);
+	}
+
+	/**
+	 * Copy file from jar to output directory.
+	 * @param jar from which file will be copied
+	 * @param fileInJar file to copy
+	 * @throws MojoFailureException if file could not be copied, inadequate
+	 *         permissions, etc.
+	 */
+	private void copyFileInJarToOutput(
+		JarFile jar, ZipEntry fileInJar
+	) throws MojoFailureException {
+		String fullPath = fileInJar.getName();
+		// Drop sourceDirectory from file path
+		String relativePath = fullPath
+			.substring(fullPath.toLowerCase(Locale.ENGLISH)
+				.indexOf(sourceDirectory.toLowerCase(Locale.ENGLISH))
+				+ sourceDirectory.length());
+		// Determine output path
+		File target = new File(outputDirectory, relativePath);
+		try {
+			if (target.exists()) {
+				if (clobber) { // Overwrite existing file
+					getLog().info(String.format(
+						"File %s already exists and clobber is set; overwriting.",
+						target.getName()
+					));
+
+					// Replace target only if contents differ
+					File tempFile = File.createTempFile(fullPath, null);
+					FileUtils.copyInputStreamToFile(
+						jar.getInputStream(fileInJar), tempFile
+					);
+					if (!FileUtils.contentEquals(tempFile, target)) {
+						FileUtils.forceDelete(target);
+						FileUtils.moveFile(tempFile, target);
+					} else {
+						tempFile.delete();
+					}
+				} else {
+					getLog().info(String.format(
+						"File %s already exists and clobber is not set; skipping.",
+						target.getName()
+					));
+				}
+			} else {
+				FileUtils.copyInputStreamToFile(
+					jar.getInputStream(fileInJar),
+					target
+				);
+			}
```

Comment:
I would prefer to break this into at least two functions because there are four layers of branching.